docs/joss_paper/paper.md

```diff
@@ -0,0 +1,155 @@
+The library provides a variety of tools including semi-Lagrangian advection operators, solvers for partial differential equations (PDEs), quadrature rules, and time steppers.
+All operators are designed to function correctly or raise compiler errors on non-orthonormal coordinate systems thanks to the static typing.
+
+## Statement of Need
+
+Plasma simulations are essential for the development of magnetic confinement fusion devices for energy production.
+The low collisionality of such plasmas make kinetic models a judicious choice.
+In particular gyrokinetic theory [@brizard2007; @krommes2012], which reduces the 6D problem to a 5D problem by removing high frequency dynamics, is a popular tool for such simulations [@garbet2010].
+Despite the reduction in dimensionality such simulations still require massively powerful high-performance computing (HPC) resources.
+For ITER-sized simulations, exascale resources would still be required.
+
+The pre-existing Gysela code [@grandgirard2016], written in Fortran, originally aimed to simulate plasma in the core region of a tokamak using semi-Lagrangian advection with a distribution function discretised in phase space on a fixed grid.
+This approach was shown to work well and allowed the study of many interesting physical phenomena [@Esteve2018;@Sarazin2021;@DifPradalier2022].
+However expanding this code to use more complex mathematical methods such as non-uniform points (vital for handling the different magnitudes of physical quantities in the core and edge regions), and increasingly complex geometries (such as D-shape geometries, geometries including both closed and open field lines, and potentially stellarator geometries) has proved to be challenging and sometimes error-prone.
+The challenges of such extensions are further amplified when trying to organise such a code for use on new GPU architectures, necessary for exascale simulations.
+This is a challenge shared by other gyrokinetic codes [@trilaksono2023].
+
+In the case of Gysela, several of the required changes would have affected a large percentage of the code base.
+This means that the effort required was comparable to a complete rewrite, without the advantages such a rewrite can bring.
+For example, we have been able to capitalise on C++'s strengths by using template programming to enforce the correctness of the implemented equations.
+A common source of error is writing equations with implicit assumptions, such as assuming an orthonormal coordinate system, or specific properties like those of a circular coordinate system.
+In Gyselalib++, equations are either expressed in tensor notation, so that they are either accurate for all geometries or do not compile, or they explicitly state their dependencies.
+C++ further enables us to add static assertions for cases with restricted applicability to prevent their misuse.
```

Comment:
Can you refer to DDC somewhere in order to explain how DDC helps to prevent many common sources of errors. This is explained in summary, but not explained in the body

Reply:
Do you have a suggestion of where?

> This is explained in summary, but not explained in the body

The summary is not equivalent to an abstract. It is just one of the sections of the paper.

Reply:
I can add something at the end of the 3rd paragraph. But if this needs doing for DDC then it probably also needs to be done for Kokkos. If I add both then I should chop something else to get back closer to the word limit. Do you see anything that you think is less important than repeating DDC?

Reply:
> The summary is not equivalent to an abstract. It is just one of the sections of the paper.

OK

> Do you have a suggestion of where?

If you can explain more in summary, that is reasonable as well.
For readers that are not very familiar with DDC, the following phrase is difficult to understand I guess
> It relies on the DDC library [@ddc] to statically type the discretisation dimensions; thus preventing many common sources of errors.


> But if this needs doing for DDC then it probably also needs to be done for Kokkos.

I think Kokkos is famous enough, so we do not need to explain more.
If you say Kokkos ensures performance portability, it is good enough

Reply:
I have added this: c72e27b8c

Reply:
Looks good